src/table.tsx

```diff
@@ -75,83 +127,207 @@ export const Table = React.forwardRef(
+                    [styles.hiddenHeadersInMobile]: hideHeadersInMobile || collapsedRowsMode,
                 })}
                 aria-label={otherProps['aria-label']}
                 aria-labelledby={otherProps['aria-labelledby']}
                 aria-describedby={otherProps['aria-describedby']}
             >
-                <thead>
-                    <Text1
-                        as="tr"
-                        medium
-                        transform="uppercase"
-                        color={vars.colors.textSecondary}
-                        wordBreak={false}
-                    >
-                        {heading.map((header, idx) => (
-                            <th
-                                scope="col"
-                                key={idx}
-                                className={classNames(
-                                    styles.cellTextAlign[getColumnTextAlign(idx)],
-                                    styles.verticalAlign[rowVerticalAlign]
-                                )}
-                                style={{minWidth: columnWidth?.[idx], width: columnWidth?.[idx]}}
-                            >
-                                {header}
-                            </th>
-                        ))}
-                    </Text1>
-                </thead>
+                {heading.length > 0 && (
+                    <thead>
+                        <Text1
+                            as="tr"
+                            medium
+                            transform="uppercase"
+                            color={vars.colors.textSecondary}
+                            wordBreak={false}
+                        >
+                            {heading.map((header, idx) => (
+                                <th
+                                    scope="col"
+                                    key={idx}
+                                    className={classNames(
+                                        styles.cellTextAlign[getColumnTextAlign(idx)],
+                                        styles.verticalAlign[rowVerticalAlign],
+                                        {
+                                            [styles.rowFirstItem]: idx === 0,
+                                            [styles.rowLastItem]:
+                                                idx === heading.length - 1 && !hasActionsColumn,
+                                        }
+                                    )}
+                                    style={{minWidth: columnWidth?.[idx], width: columnWidth?.[idx]}}
+                                >
+                                    {header}
+                                </th>
+                            ))}
+                            {hasActionsColumn && (
+                                <th>
+                                    <ScreenReaderOnly>
+                                        {/** TODO: use translated text for actions */}
```

Comment:
@aweell has requested the translations already. Once he comes back with them, I'll add them here before merging the PR

Reply:
https://jira.tid.es/browse/PRODUCTDSN-2491